### ISSUE TITLE
 101301 Secure mode: The 3DSecure authentication has failed

### ISSUE THREAD

**Issue by tomk17** [Mangopay/mangopay2-php-sdk#186]
Hy,
when i try to make a direct payment with 3dSecure on my sandbox mangopay 
the payment fails on error 101301 Secure mode: The 3DSecure authentication has failed
this is what i have in my mangopay dashboard
first payemant has succeed
33239194 | 16:06 11/09/2017 | 33239067 | 100.00 EUR | 0.00 EUR |   | 33239068 | PayIn | Succeeded
transaction detail in the dashboard : 

ID | 33239194
-- | --
Creation date | 16:06 11/09/2017
Execution date | 16:07 11/09/2017
Author | 33239067
Credited user | 33239067
Debited funds | 100.00 EUR
Credited funds | 100.00 EUR
Fees | 0.00 EUR
Credited wallet ID | 33239068
Type | PAYIN
Nature | REGULAR
Payment type | CARD
Execution type | DIRECT
Card ID | 33239193
Status | Succeeded
Result code | 000000
Result message | Success

second payment have faild
33482604 | 20:57 14/09/2017 | 33239067 | 1900.00 EUR | 0.00 EUR |   | 33239068 | PayIn | Failed | 
transaction detail in the dashboard :

ID | 33482604
-- | --
Creation date | 20:57 14/09/2017
Execution date |  
Author | 33239067
Credited user | 33239067
Debited funds | 1900.00 EUR
Credited funds | 1900.00 EUR
Fees | 0.00 EUR
Credited wallet ID | 33239068
Type | PAYIN
Nature | REGULAR
Payment type | CARD
Execution type | DIRECT
Card ID | 33482603
Status | Failed
Result code | 101301
Result message | Secure mode: The 3DSecure authentication has failed
 
I have also tried payments after the second one with lower amount (100€) and they are also failed

What happen on my transaction flow (browser) : 
When I do the payment with card   **3569990000000116** or  **3569990000000140** number for 3D secure direct payment, my brother redirect me to a blank page, then (after seconds) it comes back to my website with the error message The 3DSecure authentication has failed.
In this blank screen I should have a screen with the 3dsecurs secret input where I should put de password ″secret3″ as it used to be on my first payment 
It worked well for the first transaction (I didn't make any changes in my php code).
```
**This is my payment scripts :** 
// card registration script 
		$mangoPayApi = mangopay_api_connect();
		try{
			$cardRegister = new \MangoPay\CardRegistration();
			$cardRegister->UserId = get_mangopay_user_id_with_wp_user($user->ID);
			$cardRegister->Currency ='EUR';
			$cardRegister->CardType ='CB_VISA_MASTERCARD';
			$createdCardRegister = $mangoPayApi->CardRegistrations->Create($cardRegister);


		} catch(MangoPay\Libraries\ResponseException $e) {
				echo 'erreur : '. $e->GetCode() .' : ' .$e->GetMessage() . $e->GetErrorDetails();die;

		} catch(MangoPay\Libraries\Exception $e) {
				echo 'erreur e: '.$e->GetMessage(); die;

		}
		
		$cardRegisterId = $createdCardRegister->Id;
		// build the return URL to capture token response if browser does not support cross-domain Ajax requests
		$returnUrl = 'http' . ( isset($_SERVER['HTTPS']) ? 's' : '' ) . '://' . $_SERVER['HTTP_HOST'];
		$returnUrl .= substr($_SERVER['REQUEST_URI'], 0, strripos($_SERVER['REQUEST_URI'], '/') + 1);
		$returnUrl .= '';
 
// doing payment script 
 $mangoPayApi = mangopay_api_connect();
	try {

	    $card = $mangoPayApi->Cards->Get($card_id);
	    $payIn = new \MangoPay\PayIn();
	    $payIn->CreditedWalletId = get_wallet_id_with_wp_user($user_id);
	    $payIn->AuthorId =  get_mangopay_user_id_with_wp_user($user_id);
	    $payIn->DebitedFunds = new \MangoPay\Money();
	    $payIn->DebitedFunds->Amount = $amount * 100;
	    $payIn->DebitedFunds->Currency = 'EUR';
	    $payIn->Fees = new \MangoPay\Money();
	    $payIn->Fees->Amount = 0;
	    $payIn->Fees->Currency = 'EUR';
	    // payment type as CARD
	    $payIn->PaymentDetails = new \MangoPay\PayInPaymentDetailsCard();
   		$payIn->PaymentDetails->CardType = $card->CardType;
	    $payIn->PaymentDetails->CardId = $card->Id;
	    // execution type as DIRECT

	    $payIn->SecureMode = 'FORCE'; 
	    $payIn->SecureModeNeeded = true; 
	    
	    $payIn->ExecutionDetails = new \MangoPay\PayInExecutionDetailsDirect();	    
	    $payIn->ExecutionDetails->SecureModeReturnURL = get_site_url().'/alimentez-votre-compte/?amount='.$amount;
	    // create Pay-In
	    $createdPayIn = $mangoPayApi->PayIns->Create($payIn);
	    echo json_encode($createdPayIn);


	} catch (\MangoPay\Libraries\ResponseException $e) {
	    
	    echo '<div style="color: red;">'
	                .'\MangoPay\ResponseException: Code: ' 
	                . $e->getCode() . '<br/>Message: ' . $e->getMessage()
	                .'<br/><br/>Details: '; print_r($e->GetErrorDetails())
	        .'</div>';
	}
```
more details : 
- my website is on https mode
- it is a development  website witch is a production website copy.

Can you help me please ?
best regards
Tom

**Comment by hobailey** [comment]
Hi Tom, sorry to hear you're having problems. The 3DS cards are blocked by our PSP if the wrong 3DS password is used 3 times - although I'm saying you caused the problem, another client perhaps blocked this card between the two payments. Either way, you can use a different card (such as 3569990000000140 or 3569990000000173) and it should work fine. Either way, the problem is not related to the Mangopay API nor the PHP SDK at least :-) If you ever get this 3DS error in sandbox, it's nearly always this problem, so just try another card 👍

**Comment by tomk17** [comment]
Hi hobailey,
thank you for your answer. 
So if I understand those tow cards ( **3569990000000140 or 3569990000000173** ) are locked for my system... 
Is there any other cards number for testing the 3Dsecure mode in sandbox mode, or can you say me how to unlock this tow cards ?
regards
Tom

**Comment by hobailey** [comment]
All the 3DS testing cards are currently working, including the the two you mention - if you are having problems, I would suggest there is something wrong with your code and/or workflow. Could you contact [our support team](mangopay.com/contact) to look into this further detail (it's definitely not an SDK issue)?

**Comment by tomk17** [comment]
ok thanks for your answer ... the cards are working now...

**Comment by JbalTero** [comment]
@hobailey We have encountered this issue intermittently and we have no idea where to look on the issue. The [error guide](https://docs.mangopay.com/guide/errors) doesn't provide more info either.

You said:
> The 3DS cards are blocked by our PSP if the wrong 3DS password is used 3 times - although I'm saying you caused the problem, another client perhaps blocked this card between the two payments.

Does this mean the card will forever be blocked? or is there an expiration for the block?
If a card is blocked, is it only blocked for the particular Natural or Legal user who entered the wrong pass? Or if a card is blocked, it will be blocked within the client(who owns the user)?

**Comment by hobailey** [comment]
@JbalTero they'd be blocked for all clients, not just your client or your user. It won't be blocked forever, but may require a manual action - I'd suggest contacting the support team (as mentioned above).

**Comment by JbalTero** [comment]
@hobailey Thank you for making me understood it.
